extensions/Worker.Extensions.Http.AspNetCore/src/FunctionContext/FunctionContextExtensions.cs

```diff
@@ -27,5 +28,24 @@ public static class FunctionContextExtensions
 
             return null;
         }
+
+        /// <summary>
+        /// Gets the <see cref="HttpRequest"/> for the <see cref="FunctionContext"/>.
+        /// </summary>
+        /// <param name="context">The <see cref="FunctionContext"/>.</param>
+        /// <param name="request">The <see cref="HttpRequest"/> for the context.</param>
+        /// <returns></returns>
+        internal static bool TryGetRequest(this FunctionContext context, [NotNullWhen(true)] out HttpRequest? request)
```

Comment:
I like this pattern better than the current one for [getting the HttpContext](https://github.com/Azure/azure-functions-dotnet-worker/blob/e1ba4139510f5e6aa9eb5dae6eea5a30c9e88dad/extensions/Worker.Extensions.Http.AspNetCore/src/FunctionContext/FunctionContextExtensions.cs#L20) (above in the file). Prob for a different PR if we want to change/update it, but wanted to note that here.